build.gradle

```diff
@@ -54,18 +67,51 @@ jar {
 }
 
 shadowJar {
-    classifier = ''
+    classifier = 'unoptimised'
+}
+
+task('optimize', type: ProGuardTask, dependsOn: shadowJar) {
+    injars tasks.shadowJar.outputs.files
+    outjars "${buildDir}/libs/${project.tasks.jar.archiveName}"
+    libraryjars "${System.getProperty('java.home')}/lib/rt.jar"
```

Comment:
i don't know if this is an issue its probably fine but i have no idea

does this need to be java 8? or does it work on later ones

Reply:
IDK, everyone seems to use rt.jar (including you in the .pro you sent). I assume it works for all versions

Reply:
coolio